pkg/registry/service/port_allocator.go

```diff
@@ -0,0 +1,144 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package service
+
+import (
+	"fmt"
+	math_rand "math/rand"
+	"strconv"
+
+	"github.com/golang/glog"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/tools"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+)
+
+// The smallest size of port range we accept.
+const minPortPoolSize = 8
+
+// The base key in etc
+const baseKey = "/pools/serviceports/"
+
+type portAllocator struct {
+	portRange util.PortRange
+	pool      PoolAllocator
+}
+
+// newPortAllocator creates and initializes a new portAllocator object.
+func NewPortAllocator(portRange *util.PortRange, etcd *tools.EtcdHelper) *portAllocator {
+	rangeSize := portRange.Size
+	if rangeSize < minPortPoolSize {
+		glog.Errorf("PortAllocator requires at least %d ports", minPortPoolSize)
+		return nil
+	}
+
+	a := &portAllocator{
+		portRange: *portRange,
+	}
+
+	if etcd == nil {
```

Comment:
We can't do this as noted above - needs to be in a separate package.  You should be passing in the allocator, same as before.